### ISSUE TITLE
 Tool is silent on datalink adapter error

### ISSUE THREAD

**Issue by jacobbosch-cummins** [Equipment-and-Tool-Institute/j1939-84#1081]
We have had a few instances where there is a datalink adapter hardware issue, and the tool does not attempt to reconnect or notify the user.  Progress appears to silently "hang" for the user.  User must manually abort (usually after waiting several minutes).  No indication of errors in log file, but the CAN trace does have a "SEVERE org.etools.j1939_84.bus.RP1210Bus poll Failed to read RP1210
org.etools.j1939_84.bus.BusException: Error (142): ERR_HARDWARE_NOT_RESPONDING" message.

We recommend minimally failing loudly.  Is there an option for reconnection?

**Comment by ericthomasswenson** [comment]
Add user interface notification pop up to inform user.

Add adapter’s error notice in the log.

Review timing with developer

**Comment by markmorrisoncummins** [comment]
I experienced this recently in the part 1-2 transition. The Nexiq adapter would sometimes lose power during engine cranking, and after clicking "OK" the tool would hang on 'waiting for engine start...' even though the engine was running. Looking at the generated CAN trace, no more messages are received after the bus exception.

The stack trace led me to the poll() method of RP1210Bus, where I noticed `rp1210Executor.submit(this::poll);` is inside the `try{}` block. When an exception is caught, there doesn't appear to be any re-submission of `poll()` to the `ExecutorService`. Can `poll()` be re-submitted after an exception is caught to help robustness against brief disconnects?
